### ISSUE TITLE
ng serve didn't use the default environment.ts

### ISSUE THREAD

**Issue by lijunhaoabroad** [angular/angular-cli#16494]
# 🐞 Bug report

### Command (mark with an `x`)

- [ ] new
- [ ] build
- [X] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Description

when I run "ng serve", it uses the environment.prod.ts.
I have to change the port, "ng serve --port 4201", then it will use the default one "environment.ts"
how can I make sure "ng serve" that uses the default "environment.ts" on the default port 4200?

## 🔥 Exception or Error
<pre><code>

{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "projects",
  "projects": {
    "hello": {
      "root": "",
      "sourceRoot": "src",
      "projectType": "application",
      "prefix": "app",
      "schematics": {
        "@schematics/angular:component": {
          "styleext": "less"
        }
      },
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "dist/hello",
            "index": "src/index.html",
            "main": "src/main.ts",
            "polyfills": "src/polyfills.ts",
            "tsConfig": "src/tsconfig.app.json",
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "./node_modules/@angular/material/prebuilt-themes/pink-bluegrey.css",
              "src/styles.less"
            ],
            "scripts": []
          },
          "configurations": {
            "production": {
              "fileReplacements": [
                {
                  "replace": "src/environments/environment.ts",
                  "with": "src/environments/environment.prod.ts"
                }
              ],
              "optimization": true,
              "outputHashing": "all",
              "sourceMap": false,
              "extractCss": true,
              "namedChunks": false,
              "aot": true,
              "extractLicenses": true,
              "vendorChunk": false,
              "buildOptimizer": true,
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "2mb",
                  "maximumError": "5mb"
                }
              ]
            }
          }
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "options": {
            "browserTarget": "hello:build"
          },
          "configurations": {
            "production": {
              "browserTarget": "hello:build:production"
            }
          }
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "browserTarget": "hello:build"
          }
        },
        "test": {
          "builder": "@angular-devkit/build-angular:karma",
          "options": {
            "main": "src/test.ts",
            "polyfills": "src/polyfills.ts",
            "tsConfig": "src/tsconfig.spec.json",
            "karmaConfig": "src/karma.conf.js",
            "styles": [
              "./node_modules/@angular/material/prebuilt-themes/pink-bluegrey.css",
              "src/styles.less"
            ],
            "scripts": [],
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ]
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": [
              "src/tsconfig.app.json",
              "src/tsconfig.spec.json"
            ],
            "exclude": [
              "**/node_modules/**"
            ]
          }
        }
      }
    },
    "hello-e2e": {
      "root": "e2e/",
      "projectType": "application",
      "prefix": "",
      "architect": {
        "e2e": {
          "builder": "@angular-devkit/build-angular:protractor",
          "options": {
            "protractorConfig": "e2e/protractor.conf.js",
            "devServerTarget": "hello:serve"
          },
          "configurations": {
            "production": {
              "devServerTarget": "hello:serve:production"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": "e2e/tsconfig.e2e.json",
            "exclude": [
              "**/node_modules/**"
            ]
          }
        }
      }
    }
  },
  "defaultProject": "hello"
}

</code></pre>

## 🌍 Your Environment
<pre><code>

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/

Angular CLI: 8.3.21
Node: 13.1.0
OS: darwin x64
Angular: 8.2.14
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.803.21
@angular-devkit/build-angular     0.803.21
@angular-devkit/build-optimizer   0.803.21
@angular-devkit/build-webpack     0.803.21
@angular-devkit/core              8.3.21
@angular-devkit/schematics        8.3.21
@angular/cdk                      8.2.3
@angular/cli                      8.3.21
@angular/http                     7.2.15
@angular/material                 8.2.3
@ngtools/webpack                  8.3.21
@schematics/angular               8.3.21
@schematics/update                0.803.21
rxjs                              6.5.3
typescript                        3.5.3
webpack                           4.39.2

</code></pre>

**Anything else relevant?**

**Comment by lijunhaoabroad** [comment]
I think it is not the issue of angular. I am just confused with "*:4200" and "localhost:4200". I don't know why angular works even when port 4200 in use.

**Comment by lijunhaoabroad** [comment]
Ignore all of the above, why angular didn't prompt port 4200 is in use when *:4200 is present?
* means all interfaces including localhost, right?

**Comment by alan-agius4** [comment]
> means all interfaces including localhost, right?
No, localhost is not bound to an interface it just a loopback address. localhost has the IP address 127.0.0.1. which refers back to your own computer.

Can you try using Node 12? We official support even numbered Node versions as they are in Active LTS or will become. Odd-numbered releases (11, 13, etc.) become unsupported from Node.js after 6 months.

**Comment by lijunhaoabroad** [comment]
This could be a different issue.
I run a Go api server listening on port 4200 by using:  http.ListenAndServe(":4200", nil)
I tried to connect the backend with frontend on the same port 4200.
But somehow when I run ng serve to start my angular app, it compiled successfully without the warning  that port 4200 is in use. But it displays a blank page, angular is not working in this case.
So I run lsof -i:4200 to check what is using the port 4200.

I found out that my Go server in terminal is like this : "* :4200"
And Angular app is listening on "localhost:4200",
What does the star sign here mean here?
it means all addresses including localhost? Then why angular is not working ?
I haven't tried the Node 12, though. 
<img width="239" alt="Screen Shot 2019-12-28 at 16 55 39" src="https://user-images.githubusercontent.com/15603606/71546071-28d1f080-2993-11ea-8688-d223695dbbee.png">


Thank you!
